src/app/(auth)/reset-password/page.tsx

```diff
@@ -0,0 +1,68 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/navigation';
+import TextInput from '@/components/TextInput/index';
+import { H1, H4 } from '@/styles/text';
+import { QuestionsDiv } from '@/app/(auth)/styles';
+import BigButton from '@/components/BigButton';
+import supabase from '../../../api/supabase/createClient';
+
+export default function ResetPassword() {
+  const [newPassword, setNewPassword] = useState('');
+  const [newPassword2, setNewPassword2] = useState('');
+  const [canReset, setCanReset] = useState(false);
+  const { push } = useRouter();
+  useEffect(() => {
+    supabase.auth.onAuthStateChange(async event => {
+      if (event === 'PASSWORD_RECOVERY') {
+        setCanReset(true);
+      }
+    });
+  }, []);
+
+  const resetPassword = async () => {
+    const { error } = await supabase.auth.updateUser({ password: newPassword });
+    if (error) {
+      throw new Error(
+        `An error occurred trying to reset password: ${error.message}`,
+      );
+    }
+    const { error: signOutError } = await supabase.auth.signOut();
+    if (signOutError) {
+      throw new Error(
+        `An error occurred trying to sign out: ${signOutError.message}`,
+      );
+    }
```

Comment:
I've been wondering: should we add a `src/api/supabase/auth/...` for supabase auth calls? Could declutter some error handling if we'll be repeating these calls.

Reply:
oooo icic, i'm not entirely sure what this means, but let's discuss today? 

the only supabase i'm making throughout auth is `'@/api/supabase/createClient'`